framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -5998,110 +6029,146 @@ VkResult VulkanReplayConsumerBase::OverrideCreatePipelineCache(
     GFXRECON_UNREFERENCED_PARAMETER(original_result);
 
     GFXRECON_ASSERT((device_info != nullptr) && (pCreateInfo != nullptr) && (pPipelineCache != nullptr) &&
-                    (pPipelineCache->GetHandlePointer() != nullptr));
+                    (pPipelineCache->GetHandlePointer() != nullptr) && (pCreateInfo->GetPointer() != nullptr));
 
-    auto replay_create_info = pCreateInfo->GetPointer();
-    GFXRECON_ASSERT(replay_create_info != nullptr);
+    VkPipelineCacheCreateInfo override_create_info = *pCreateInfo->GetPointer();
 
-    if (options_.omit_pipeline_cache_data)
+    // If pipeline cache must be loaded from file
+
+    std::vector<char> pipelineCacheData;
+    if (!options_.load_pipeline_cache_filename.empty())
     {
-        // Make a shallow copy of the create info structure and clear the cache data.
-        VkPipelineCacheCreateInfo override_create_info = (*replay_create_info);
+        LoadPipelineCache(*pPipelineCache->GetPointer(), pipelineCacheData);
 
-        if (replay_create_info->initialDataSize != 0)
+        if (!pipelineCacheData.empty())
+        {
+            override_create_info.initialDataSize = pipelineCacheData.size();
+            override_create_info.pInitialData    = pipelineCacheData.data();
+        }
+        else
+        {
+            // If capture data could not be loaded from file, do not fail, just do not use pipeline cache data. We want
+            // this behaviour so that a cache can be created incrementally by loading the partial cache from the
+            // previous run and feeding it to create the next cache with more data until everything is built.
+            override_create_info.initialDataSize = 0;
+            override_create_info.pInitialData    = nullptr;
+        }
+    }
+
+    // If pipeline cache must not be loaded
+
+    else if (options_.omit_pipeline_cache_data)
```

Comment:
It seems to me that if the omit is present, we shouldn't do any pipeline cache loading.  Is this really the way we want to behave?

Reply:
agree with you, omit means omit. looking into it

Reply:
to me it looks it matches the previous behavior in this case. I understand the intention as "omit any data" - a cache is created in any case, but might be empty. we set member `omitted_pipeline_cache_data_` to true in case we actually omitted anything

Reply:
But what if the cache isn't empty.  We're saying if the user selects load cache and omit cache, the load overrides the omit.  Seems wrong to me.  Maybe someone else can comment.

Reply:
ok, took me a bot longer. went back to that part and rewrote to fully respect `options_.omit_pipeline_cache_data`. no data will be loaded if this is set and "load cache" option cannot override.